addons/mail/static/src/models/thread_cache/thread_cache.js

```diff
@@ -26,18 +26,21 @@ function factory(dependencies) {
             this.update({ isLoadingMore: true });
             const messageIds = this.fetchedMessages.map(message => message.id);
             const limit = 30;
-            const fetchedMessages = await this.async(() => this._loadMessages({
-                extraDomain: [['id', '<', Math.min(...messageIds)]],
-                limit,
-            }));
-            this.update({ isLoadingMore: false });
-            if (fetchedMessages.length < limit) {
-                this.update({ isAllHistoryLoaded: true });
-            }
-            for (const threadView of this.threadViews) {
-                threadView.addComponentHint('more-messages-loaded', { fetchedMessages });
+            try {
+                const fetchedMessages = await this.async(() => this._loadMessages({
+                    extraDomain: [['id', '<', Math.min(...messageIds)]],
+                    limit,
+                }));
+                if (fetchedMessages.length < limit) {
+                    this.update({ isAllHistoryLoaded: true });
+                }
+                for (const threadView of this.threadViews) {
+                    threadView.addComponentHint('more-messages-loaded', { fetchedMessages });
+                }
+                return fetchedMessages;
+            } finally {
+                this.update({ isLoadingMore: false });
```

Comment:
I would have preferred limiting `try/catch/finally` on only block of code that could fail (here the actual fetching of messages). Also method return being either list of messages or `undefined` show something is off, because `undefined` happens when there's an error, but that case should be managed with error paradigm (`try/catch` & `throw`) rather than sentinel values.

This issue was already there before this PR, so I wouldn't make the change in this PR and instead clean this later.

Reply:
Does it actually return undefined? This is a try with no catch, so isn't the exception propagated to caller? I have some troubles to understand try/catch with await, I should probably make some tests because it doesn't seem to work as I would expect.

Reply:
Ignore my comment but I guess comment from Alex is still open here. Would be best to limit the scope of the try to only the RPC.